fearless/Common/PurchaseProvider/RampProvider.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+
+final class RampProvider: PurchaseProviderProtocol {
+    static let pubToken = "3quzr4e6wdyccndec8jzjebzar5kxxzfy2f3us5k"
+    static let baseUrlString = "https://buy.ramp.network/"
+
+    private var appName: String?
+    private var logoUrl: URL?
+
+    func with(appName: String) -> Self {
+        self.appName = appName
+        return self
+    }
+
+    func with(logoUrl: URL) -> Self {
+        self.logoUrl = logoUrl
+        return self
+    }
+
+    func buildPurchaseAction(for chain: Chain,
+                             assetId: WalletAssetId?,
+                             address: String) -> [PurchaseAction] {
+        let optionUrl: URL?
+
+        switch chain {
+        case .polkadot:
+            optionUrl = buildURLForToken("DOT", address: address)
+        case .kusama:
+            optionUrl = nil
+        case .westend:
+            optionUrl = nil
+        }
+
+        if let url = optionUrl {
+            let action = PurchaseAction(title: "Ramp", url: url)
+            return [action]
+        } else {
+            return []
+        }
+    }
+
+    private func buildURLForToken(_ token: String, address: String) -> URL? {
+        var components = URLComponents(string: Self.baseUrlString)
+
+        var queryItems = [
+            URLQueryItem(name: "swapAsset", value: token),
+            URLQueryItem(name: "userAddress", value: address),
+            URLQueryItem(name: "hostApiKey", value: Self.pubToken)
```

Comment:
in the future, if we will want to support Staging envs for payment providers, this probably should be changed

Reply:
Actually not, as we can see Self.pubToken and Self.baseUrl are static vars and can be wrapped with preprocessing macros to have different values for different environments